### PR TITLE
udev: small cleanups

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/udev/udev-builtin.c [hunk 1]
```diff
@@ -102,18 +102,19 @@ enum udev_builtin_cmd udev_builtin_lookup(const char *command) {
...
+        r = strv_split_extract(&argv, command, NULL, EXTRACT_QUOTES|EXTRACT_RELAX|EXTRACT_CUNESCAPE|EXTRACT_CUNESCAPE_RELAX);
+        if (r < 0)
+                return r;
+
         /* we need '0' here to reset the internal state */
         optind = 0;
-        strscpy(arg, sizeof(arg), command);
-        udev_build_argv(arg, &argc, argv);
-        return builtins[cmd]->cmd(dev, argc, argv, test);
+        return builtins[cmd]->cmd(dev, strv_length(argv), argv, test);
```
Comment (keszybz) [inline src/udev/udev-builtin.c:116]: Does `strv_split_extract` implement the same semantics? Do we have some tests for this?

**Comment by keszybz** [comment]
The code looks good, but (as always), I worry about unintentional changes. I don't doubt that if the new code behaves differently in any way, then the new behaviour is better, but if there *are* any changes, we should at least document them.

**Comment by poettering** [comment]
looks good to me too, but i am a bit afraid of this without tests... any chance you can add a test for this somehow that validates compat with old versions, i.e. that some cmdlines are parsed the way expected with the old code with the new code too?

**Comment by yuwata** [comment]
@keszybz and @poettering, thank you for the review.
I've checked this by running udev-test.pl, and all the tests listed in the script pass. I thought the script tests this well. But if you think the tests are not enough, then I will try to add more.

**Comment by keszybz** [comment]
I added `log_info("%s: [%s]", __func__, command);` to `udev_builtin_run`, and I run `udev-test`. It seems that udev_builtin_run is only called twice:
```
udev_builtin_run: [path_id]
udev_builtin_run: [path_id]
```
I think we need some more direct testing for this.

**Comment by yuwata** [comment]
It seems that `udev_event_spawn()` is tested more but udev-test.pl. Both `udev_builtin_run()` and `udev_event_spawn()` used the same command line parser. So, I thought we can safely replace `udev_build_argv()`.
But, anyway, I will try to add more tests, especially for `udev_builtin_run()`.

**Comment by yuwata** [comment]
Hmm, it is difficult for me to add rules for builtin to test the argv builder.
I've manually tested this with 'udevadm test-builtin', and it seems works fine.

**Comment by yuwata** [comment]
Updated.
- introduce `strv_split_full()` which can take some flags.
- add a test to compare `udev_build_argv()` and `strv_split_full()`. It seems both functions provides almost equivalent result. One difference is that `udev_build_argv()` does not handle white spaces on the head.

If this is accepted, then I will drop `udev_build_argv()` and test-udev-build-argv.c later, as the function is now only used by the new test.

PTAL. Thank you.

**Comment by keszybz** [comment]
Looks good, but bionic-amd64 failed:
```
FAILED TESTS:  test/TEST-17-UDEV-WANTS test/TEST-18-FAILUREACTION
autopkgtest [15:42:56]: test upstream: -----------------------]
upstream             FAIL non-zero exit status 1
```

I'll restart the test.

**Comment by yuwata** [comment]
TEST-02-CRYPTSETUP timeouts.

**Comment by keszybz** [comment]
Restarted again.

**Comment by yuwata** [comment]
TEST-02-CRYPTSETUP timeouts again...

**Pull request merged**